distributed/client.py

```diff
@@ -1367,7 +1367,13 @@ def wait_for_workers(self, n_workers=0, timeout=None):
             Time in seconds after which to raise a
             ``dask.distributed.TimeoutError``
         """
-        return self.sync(self._wait_for_workers, n_workers, timeout=timeout)
+        try:
+            self.scale(n_workers)
```

Comment:
I think adding this would quite dramatically change behavior of this API. Is this intentional? https://github.com/dask/distributed/issues/6346 doesn't mention this

Reply:
You're right, this is a mistake. 

I added this to get test_n_workers in test_ssh.py to work. The test looks like it starts a cluster with 2 workers and then waits for 4 to be running, which never happened so I scaled up to 4 workers. I don't understand how the code was, before I added the call to scale, functionally any different to main branch. I also don't understand how that test passes on main if it is waiting for more workers than it has, wouldn't it be waiting forever?

Reply:
I have removed this call to scale. The problem I was seeing was due to cluster.scheduler_info not updating. I thought it was always updated, but it turns out it only updates when certain commands are sent to the scheduler, including the call to add workers. Calling scale adds workers and updates the scheduler_info, so it fixed the bugs I was seeing. Now I manually update the scheduler_info, fixing the original issue and meaning this call to scale is no longer necessary (and as you said, was a dramatic change of behaviour anyway). 

Thanks for your comment, it forced me to look more deeply into the problem and find a better solution